app/src/main/java/tech/ula/ui/AppListFragment.kt

```diff
@@ -104,11 +114,17 @@ class AppListFragment : Fragment() {
                     return@OnItemClickListener
                 }
 
-                val serviceIntent = Intent(activityContext, ServerService::class.java)
-                        .putExtra("type", "startApp")
-                        .putExtra("app", selectedApp)
-                        .putExtra("serviceType", "ssh") // TODO update this dynamically based on user preference
-                activityContext.startService(serviceIntent)
+                val preferredClient = appListViewModel.getAppServiceTypePreference(appList[position])
+                if (preferredClient.isEmpty()) {
```

Comment:
Nice use of the null object pattern. I didn't actually know you could call `isEmpty()` on strings like that.